### PR TITLE
Update `flutter_lints` to v2

### PR DESCRIPTION
**Pull request by Widcket**
### Description

This PR upgrades the `flutter_lints` library to v2 in all packages, and fixes the issues flagged by `flutter analyze`. Also, the rule `package_api_docs` was enabled.

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

### CODE REVIEW

**Review by frederikprijck — APPROVED** [review]

File: auth0_flutter/example/lib/main.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@ import 'dart:async';
 
 import 'package:flutter/material.dart';
 import 'package:flutter_dotenv/flutter_dotenv.dart';
+// ignore: depend_on_referenced_packages
```
Comment (stevehobbsdev) [inline auth0_flutter/example/lib/main.dart:5]: Why is this exception needed?

File: auth0_flutter/lib/src/authentication_api.dart [hunk 2]
```diff
@@ -3,10 +3,10 @@ import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interfac
 /// An interface for calling some of the endpoints in [Auth0's Authentication API](https://auth0.com/docs/api/authentication).
 ///
 /// This class presents building blocks for doing more fine-grained authentication with Auth0 using Username and Password login. Unlike
-/// [WebAuthentication], these do **not** use [Auth0 Universal Login](https://auth0.com/docs/authenticate/login/auth0-universal-login) (the recommended way of doing authentication),
+/// `WebAuthentication`, these do **not** use [Auth0 Universal Login](https://auth0.com/docs/authenticate/login/auth0-universal-login) (the recommended way of doing authentication),
```
Comment (stevehobbsdev) [inline auth0_flutter/lib/src/authentication_api.dart:6]: What's the reason for this change?

Comment (Widcket) [reply auth0_flutter/lib/src/authentication_api.dart:6]: The linter complains that the types referenced are not defined on this file.

File: auth0_flutter/example/lib/main.dart (hunk 1 above)
Comment (Widcket) [reply auth0_flutter/example/lib/main.dart:5]: The linter complains that the driver package is not in the app's dependencies (it's on dev_dependencies).

Comment (frederikprijck) [reply auth0_flutter/example/lib/main.dart:5]: We could move it into the dependencies I believe, because it's actually used in the app itself so it does feel like a dependency instead of a devDependency to some extend.
